contracts/examples/guards/BaseGuard.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.7.0 <0.9.0;
+import {IERC165} from "../../interfaces/IERC165.sol";
+import {BaseTransactionGuard, Guard} from "../../base/GuardManager.sol";
+import {BaseModuleGuard, IModuleGuard} from "../../base/ModuleManager.sol";
+
+/**
+ * @title BaseGuard - Inherits BaseTransactionGuard and BaseModuleGuard.
+ */
+abstract contract BaseGuard is BaseTransactionGuard, BaseModuleGuard {
+    /**
+     * @notice Checks if the contract supports an interface.
+     * @param interfaceId The interface identifier.
+     * @return True if the interfaceId matches the interfaceId of Guard/IModuleGuard/IERC165, false otherwise.
+     */
+    function supportsInterface(bytes4 interfaceId) external view virtual override(BaseTransactionGuard, BaseModuleGuard) returns (bool) {
+        return
+            interfaceId == type(Guard).interfaceId || // 0xe6d7a83a
+            interfaceId == type(IModuleGuard).interfaceId || // 0xe1ab3a1a
```

Comment:
Hmm, this looks wrong to me, I would expect this to be:

```suggestion
            interfaceId == type(ITransactionGuard).interfaceId || // 0xe6d7a83a
            interfaceId == type(IModuleGuard).interfaceId || // 0xe1ab3a1a
```

Reply:
Oh I see, there is a bit of inconsistent naming with the `Guard` types - can we make them more consistent? Maybe `ITransactionGuard` and `IModuleGuard` or `TransactionGuard` and `ModuleGuard`? Or (although not a huge fan of this) either `Guard` and `ModuleGuard` or `IGuard` and `IModuleGuard`?

Reply:
We have `ITransactionGuard` and `IModuleGuard`